src/oss-detect-cryptography/DetectCryptographyTool.cs

```diff
@@ -107,55 +108,79 @@ private static async Task Main(string[] args)
+                                foreach (var tag in implementations)
+                                {
+                                    sb.AppendLine(Bright.Blue($" * {tag}"));
+                                }
                             }
                             else
                             {
-                                sb.AppendLine($"[ ] {target} - This software package does NOT have a high-density of cryptographic operators.");
+                                sb.AppendLine(Bright.Black("  No implementations found."));
                             }
 
-                            if (otherTags.Contains("Cryptography.GenericImplementation.CryptographicWords"))
+                            sb.AppendLine();
+                            sb.AppendLine(Red("Cryptographic Library References:"));
+                            var references = results.SelectMany(r => r.Issue.Rule.Tags ?? new List<string>())
+                                                    .Distinct()
+                                                    .Where(t => t.StartsWith("Cryptography.Reference."))
+                                                    .Select(t => t.Replace("Cryptography.Reference.", ""))
+                                                    .OrderBy(s => s);
+
+                            if (references.Any())
                             {
-                                sb.AppendLine($"[X] {target} - This software package contains words that suggest cryptography.");
+                                foreach (var tag in references)
+                                {
+                                    sb.AppendLine(Bright.Red($" * {tag}"));
+                                }
                             }
                             else
                             {
-                                sb.AppendLine($"[ ] {target} - This software package does NOT contains words that suggest cryptography.");
+                                sb.AppendLine(Bright.Black("  No library references found."));
                             }
 
-                            foreach (var t in otherTags)
+                            sb.AppendLine();
+                            sb.AppendLine(Green("Other Cryptographic Characteristics:"));
+                            var characteristics = results.SelectMany(r => r.Issue.Rule.Tags ?? new List<string>())
+                                                         .Distinct()
+                                                         .Where(t => t.Contains("Crypto", StringComparison.InvariantCultureIgnoreCase)&&
+                                                                     !t.StartsWith("Cryptography.Implementation.") &&
+                                                                     !t.StartsWith("Cryptography.Reference."))
+                                                         .Select(t => t.Replace("Cryptography.", ""))
```

Comment:
If I'm not misunderstanding wouldn't this line here get rid of the "Cryptography" portion of the tag that you just found with the Where command.  Why modify the tags?

Reply:
Just to make the output cleaner -- a bunch of tags in ApplicationInspector have `Cryptography.` prefixes, which just clutters the screen.